src/main/java/net/laboulangerie/laboulangeriemmo/core/abilities/thehunter/FireBow.java

```diff
@@ -77,15 +82,21 @@ else if (block != null)
         FireArrow.fireArrow.remove(fireArrow);
     }
 
-    private static void onEntityHit(FireArrow fireArrow, Entity entity) {
-        final int level = fireArrow.getAbilityLevel();
+    private static void onEntityHit(FireArrow fireArrow, Entity entity)  {
+        if (!(fireArrow.getShooter().getInventory().getItemInMainHand().getItemMeta().hasEnchant(Enchantment.ARROW_FIRE))) {
```

Comment:
Doesn't work, player can change its item in hand before the arrow hits an entity